src/main-web.ts

```diff
@@ -126,179 +124,195 @@ function render_pointer_history(ctx: CanvasRenderingContext2D) {
-        const xb = pointB.x + (bodyB ? bodyB.position.x : 0)
-        const yb = pointB.y + (bodyB ? bodyB.position.y : 0)
+    render_debug_matter_query(ctx: CanvasRenderingContext2D) {
+        ctx.fillStyle = "rgba(255, 0, 0, 0.3)"
+        for (const query of this.naubino.debug_matter_queries) {
+            ctx.beginPath()
+            ctx.moveTo((<any>query)[0].x, (<any>query)[0].y)
+            for (const p of (<Array<Vector>>query).slice(1)) {
+                ctx.lineTo(p.x, p.y)
+            }
+            ctx.closePath()
+            ctx.fill()
+        }
+        this.naubino.debug_matter_queries = []
+    }
 
-        ctx.save()
-        if (userData) {
-            switch (userData.type) {
-                case "ArenaMode.CenterJoint":
-                    ctx.setLineDash([1, 5])
-                    ctx.strokeStyle = "lightgray"
-                    ctx.lineWidth = 1
-                    break
-                case "NaubJoint":
-                    ctx.strokeStyle = "gray"
-                    ctx.lineWidth = 1
-                    break
-                default:
-                    ctx.strokeStyle = "red"
-                    ctx.lineWidth = 2
+    render_matter(ctx: CanvasRenderingContext2D) {
+        const world = this.naubino.engine.world
+        for (const constraint of world.constraints) {
+            const { pointA, pointB, bodyA, bodyB, userData } = constraint
+            const xa = pointA.x + (bodyA ? bodyA.position.x : 0)
+            const ya = pointA.y + (bodyA ? bodyA.position.y : 0)
+            const xb = pointB.x + (bodyB ? bodyB.position.x : 0)
+            const yb = pointB.y + (bodyB ? bodyB.position.y : 0)
+
+            ctx.save()
+            if (userData) {
+                switch (userData.type) {
+                    case "ArenaMode.CenterJoint":
+                        ctx.setLineDash([1, 5])
+                        ctx.strokeStyle = "lightgray"
+                        ctx.lineWidth = 1
+                        break
+                    case "NaubJoint":
+                        ctx.strokeStyle = "gray"
+                        ctx.lineWidth = 1
+                        break
+                    default:
+                        ctx.strokeStyle = "red"
+                        ctx.lineWidth = 2
+                }
+            } else {
+                ctx.strokeStyle = "red"
+                ctx.lineWidth = 2
             }
-        } else {
-            ctx.strokeStyle = "red"
-            ctx.lineWidth = 2
+            ctx.beginPath()
+            ctx.moveTo(xa, ya)
+            ctx.lineTo(xb, yb)
+            ctx.closePath()
+            ctx.stroke()
+            ctx.restore()
         }
-        ctx.beginPath()
-        ctx.moveTo(xa, ya)
-        ctx.lineTo(xb, yb)
-        ctx.closePath()
-        ctx.stroke()
-        ctx.restore()
     }
-}
\ No newline at end of file
+}
+
+let canvas: HTMLCanvasElement
+
+const GAME = new Game();
```

Comment:
canvas should be some parameter to Game instead of a global

Reply:
yapp, I only changed as much as was needed for the feature, lets go ahead and put it all into Game